### PR TITLE
[impeller] Wireup vertex descriptor format for vulkan

### PR DESCRIPTION
**Pull request by iskakaushik**

### CODE REVIEW

File: impeller/renderer/backend/vulkan/allocator_vk.cc [hunk 1]
```diff
@@ -121,7 +121,10 @@ std::shared_ptr<DeviceBuffer> AllocatorVK::OnCreateBuffer(
   // StorageMode.
   auto buffer_create_info = static_cast<vk::BufferCreateInfo::NativeType>(
       vk::BufferCreateInfo()
-          .setUsage(vk::BufferUsageFlagBits::eStorageBuffer |
+          .setUsage(vk::BufferUsageFlagBits::eVertexBuffer |
+                    vk::BufferUsageFlagBits::eIndexBuffer |
+                    vk::BufferUsageFlagBits::eUniformBuffer |
+                    vk::BufferUsageFlagBits::eStorageBuffer |
```
Comment (chinmaygarde) [inline impeller/renderer/backend/vulkan/allocator_vk.cc:127]: You can get rid of eStorageBuffer as that is not a usage we have in Impeller.

File: impeller/renderer/backend/vulkan/device_buffer_vk.h [hunk 2]
```diff
@@ -45,6 +45,8 @@ class DeviceBufferVK final : public DeviceBuffer,
   // |DeviceBuffer|
   ~DeviceBufferVK() override;
 
+  DeviceBufferAllocationVK* GetAllocation() const;
```
Comment (chinmaygarde) [inline impeller/renderer/backend/vulkan/device_buffer_vk.h:48]: Nit: `GetVKDeviceBufferAllocation` perhaps?

File: impeller/renderer/backend/vulkan/pipeline_vk.cc [hunk 3]
```diff
@@ -6,22 +6,37 @@
...
+      pipeline_ && render_pass_ && pipeline_layout_ && descriptor_set_layout_;
 }
 
 bool PipelineCreateInfoVK::IsValid() const {
   return is_valid_;
 }
 
-vk::UniquePipeline PipelineCreateInfoVK::GetPipeline() {
-  return std::move(pipeline_);
+vk::Pipeline PipelineCreateInfoVK::GetPipeline() const {
```
Comment (chinmaygarde) [inline impeller/renderer/backend/vulkan/pipeline_vk.cc:26]: `GetVKPipeline` perhaps?

File: impeller/renderer/backend/vulkan/vertex_descriptor_vk.cc [hunk 4]
```diff
@@ -6,8 +6,160 @@
 
 namespace impeller {
 
-VertexDescriptorVK::VertexDescriptorVK() = default;
+vk::Format ToVertexDescriptorFormat(const ShaderStageIOSlot& input) {
```
Comment (chinmaygarde) [inline impeller/renderer/backend/vulkan/vertex_descriptor_vk.cc:9]: static method?

**Review by chinmaygarde — APPROVED** [review]

**Comment by iskakaushik** [comment]
I have a PR coming up very shortly, will fix up all these comments as part of that.

**Pull request merged**